src/qinterface/gates.cpp

```diff
@@ -12,6 +12,23 @@
 
 #include "qinterface.hpp"
 
+#define C_SQRT1_2 complex(M_SQRT1_2, ZERO_R1)
+#define ONE_PLUS_I_DIV_2 complex(ONE_R1 / 2, ONE_R1 / 2)
+#define ONE_MINUS_I_DIV_2 complex(ONE_R1 / 2, -ONE_R1 / 2)
+
+#define GATE_1_BIT(gate, mtrx00, mtrx01, mtrx10, mtrx11)                                                               \
+    void QInterface::gate(bitLenInt qubit)                                                                             \
+    {                                                                                                                  \
+        const complex mtrx[4] = { mtrx00, mtrx01, mtrx10, mtrx11 };                                                    \
+        ApplySingleBit(mtrx, true, qubit);                                                                             \
+    }
+
+#define GATE_1_PHASE(gate, topLeft, bottomRight)                                                                       \
+    void QInterface::gate(bitLenInt qubit) { ApplySinglePhase(topLeft, bottomRight, false, qubit); }
+
+#define GATE_1_INVERT(gate, topRight, bottomLeft)                                                                      \
+    void QInterface::gate(bitLenInt qubit) { ApplySingleInvert(topRight, bottomLeft, false, qubit); }
```

Comment:
How about this?

```c++
#define GATE_1_BIT(mtrx00, mtrx01, mtrx10, mtrx11)                                                               \
    {                                                                                                                  \
        const complex mtrx[4] = { mtrx00, mtrx01, mtrx10, mtrx11 };                                                    \
        ApplySingleBit(mtrx, true, qubit);                                                                             \
    }

...
void QInterface::H(bitLenInt qubit)
    GATE_1_BIT(C_SQRT1_2, C_SQRT1_2, C_SQRT1_2, -C_SQRT1_2);

// Works for parameterized variants too.
void QInterface::ApplySinglePhase(
        const complex topLeft, const complex bottomRight, bool doCalcNorm, bitLenInt qubitIndex)
    GATE_1_BIT( topLeft, ZERO_CMPLX, ZERO_CMPLX, bottomRight);
```

That preserves the ease of finding the implementation, keeps things like doxygen and ctags happy, etc.

I won't push hard against what it is right now, but consider if you like the above a little better and make the change as you see fit.

Reply:
Just to be absolutely clear - the above is offered as an alternative that some people (and tools) find easier to read.  The current implementation is acceptable, if the above isn't preferred by the author.

Reply:
Let me check whether the Doxygen can find the prototypes. If not, I'll take your suggestion. Otherwise, I've found there's still enough predictable entropy for locating the definitions by string search.

Reply:
For the record, the Doxygen _can_ find them, I think due to the header definitions. So, I prefer the terser version, but we'll revisit this if it becomes inconvenient.